File: dagger-mvp/src/main/java/be/vergauwen/simon/daggermvp/ui/TestComponent.java
package be.vergauwen.simon.daggermvp.ui;

import be.vergauwen.simon.daggermvp.di.ActivityScope;
import be.vergauwen.simon.daggermvp.di.ApplicationComponent;
import be.vergauwen.simon.himura.MVPContract;
import dagger.Component;

@ActivityScope
@Component(dependencies = ApplicationComponent.class)
public interface TestComponent extends MVPContract.Component<TestPresenter> {
    TestPresenter presenter();
}

File: kotlin-dagger-mvp/src/main/kotlin/be/vergauwen/simon/kotlindaggermvp/ui/TestComponent.kt
package be.vergauwen.simon.kotlindaggermvp.ui

import be.vergauwen.simon.kotlindaggermvp.di.ActivityScope
import be.vergauwen.simon.kotlindaggermvp.di.ApplicationComponent
import dagger.Component

@ActivityScope
@Component(dependencies = arrayOf(ApplicationComponent::class))
interface TestComponent : TestContract.Component<TestContract.View, TestPresenter> {
  override fun presenter(): TestPresenter
}